spec/factories/requests.rb

```diff
@@ -10,5 +10,20 @@
     decision     { :undecided }
 
     workflow
+
+    trait :with_context do
+      after(:create) do |obj|
+        obj.update_attributes(:context =>
+          {
```

Comment:
@bzwei 
We have a spec helper for this
https://github.com/ManageIQ/catalog-api/blob/148f26523bbd46706eccdeb6ba35c4c90a90d8f9/spec/support/service_spec_helper.rb#L17

I have done more cleanup in this PR for catalog
https://github.com/ManageIQ/catalog-api/pull/228

Reply:
In your example spec helper why does the `original_url` appear twice?

Reply:
The original_url is supposed to be part of the "Request" and not the headers. Initially I had added it to the headers and we cleaned it up so that its only part of the request.